src/lib.rs

```diff
@@ -0,0 +1,101 @@
+/*###############################################################
+###                                                             #
+###                        KVStore Code                         #
+###                                                             #
+#################################################################
+*/
+
+use std::collections::HashMap;
+use std::io;
+
+// Define a public struct named KVStore to represent a key-value store.
+// NOTE: This code is responsible for a simple REPL (Read-Eval-Print Loop) 
+//that supports basic commands like SET, GET, DELETE, CONTAINS, and QUIT.
+
+pub struct KVStore {
+    // The store field is a HashMap that associates strings (keys) with strings (values).
+    store: HashMap<String, String>,
+}
+
+// Implement methods for the KVStore struct.
+impl KVStore {
+    // Constructor for KVStore, creating a new instance with an empty HashMap.
+    pub fn new() -> Self {
+        KVStore {
+            store: HashMap::new(),
+        }
+    }
+
+    // Insert a key-value pair into the store, returning the previous value if the key already existed.
+    pub fn set(&mut self, key: String, value: String) -> Option<String> {
+        self.store.insert(key, value)
```

Comment:
Let's add logging for the initial phases. `info` logs for everything that is getting stored. Not part of REPL.